code/modules/materials/materials.dm

```diff
@@ -0,0 +1,198 @@
+/datum/material
+	var/name = "material"
+	var/color = "#FFFFFF"
+	var/force_multiplier = 1
+	var/health_multiplier = 1
+	var/prefix = "material"
+	var/starting_material = 0
+	var/atom/my_atom
+	var/can_process = 0
+	var/conduction = 0
+	var/obj/item/stack/default_stack
+
+/datum/material/proc/fire_act()
+	return
+
+/datum/material/proc/throw_impact(atom/hit_atom)
+	return
+
+/datum/material/proc/init_material(var/atom/A)
+	return
+
+/datum/material/iron
+	name = "Iron"
+	color = "#908D8D"
+	force_multiplier = 1
+	health_multiplier = 1
+	prefix = "iron"
+	conduction = 1
+	default_stack = /obj/item/stack/sheet/metal
+
+/datum/material/gold
+	name = "Gold"
+	color = "" // Handled by the material init. proc.
+	force_multiplier = 0.25
+	health_multiplier = 0.25 // this aint digmineconstructcraft buddy
+	prefix = "gold"
+	conduction = 1
+	default_stack = /obj/item/stack/sheet/mineral/gold
+
+
+/datum/material/diamond
+	name = "Diamond"
+	color = "#BFFFFF"
+	force_multiplier = 2
+	health_multiplier = 4
+	prefix = "diamond"
+	default_stack = /obj/item/stack/sheet/mineral/diamond
+/datum/material/glass
+	name = "Glass"
+	color = "#FFFFFF"
+	force_multiplier = 0.5
+	health_multiplier = 0.5
+	prefix = "glass"
+	default_stack = /obj/item/stack/sheet/glass
+/datum/material/glass/throw_impact(atom/hit_atom)
+	if(prob(50))
+		my_atom.visible_message("[my_atom] shatters!")
+		qdel(my_atom)
+	return
+
+/datum/material/sandstone
+	name = "Sandstone"
+	color = "#E7C88F"
+	force_multiplier = 1
+	health_multiplier = 1
+	prefix = "sandstone"
+	default_stack = /obj/item/stack/sheet/mineral/sandstone
+/datum/material/uranium
+	name = "Uranium"
+	color = "#39C639"
+	force_multiplier = 3
+	health_multiplier = 5
+	prefix = "uranium"
+	var/active_tick = 0
+	default_stack = /obj/item/stack/sheet/mineral/uranium
+/datum/material/uranium/process()
+	if(active_tick >= 15)
+		my_atom.visible_message("[my_atom] glows green.")
+		for(var/mob/living/L in range(3,my_atom))
+			L.irradiate(12)
+		active_tick = 0
+		return
+	active_tick++
+	return
+
+/datum/material/plasma
+	name = "Plasma"
+	color = "#E122D9"
+	force_multiplier = 1
+	health_multiplier = 1
+	prefix = "plasma"
+	default_stack = /obj/item/stack/sheet/mineral/plasma
```

Comment:
Things made from *solid* plasma are flammable. The liquid is not. The ore is not. The ingot is not, but structures (tiles, statues, walls, airlocks, doors) are all flammable.